pkg/processor/sourceprocessor/source_category_filler.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2021 Sumo Logic, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sourceprocessor
+
+import (
+	"fmt"
+	"strings"
+
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+// sourceCategoryFiller adds source category attribute to a collection of attributes.
+type sourceCategoryFiller struct {
+	attributeName      string
+	valueTemplate      string
+	templateAttributes []string
+	prefix             string
+	dashReplacement    string
+	annotationPrefix   string
+}
+
+// newSourceCategoryFiller creates a new sourceCategoryFiller.
+func newSourceCategoryFiller(
+	attributeName string,
+	cfg *Config,
+) sourceCategoryFiller {
+
+	valueTemplate := cfg.SourceCategory
+	templateAttributes := extractTemplateAttributes(valueTemplate)
+
+	return sourceCategoryFiller{
+		attributeName:      attributeName,
+		valueTemplate:      valueTemplate,
+		templateAttributes: templateAttributes,
+		prefix:             cfg.SourceCategoryPrefix,
+		dashReplacement:    cfg.SourceCategoryReplaceDash,
+		annotationPrefix:   cfg.AnnotationPrefix,
+	}
+}
+
+func extractTemplateAttributes(template string) []string {
+	attributes := make([]string, 0)
+	attributeMatches := formatRegex.FindAllStringSubmatch(template, -1)
```

Comment:
```suggestion
	attributeMatches := formatRegex.FindAllStringSubmatch(template, -1)
	attributes := make([]string, 0, len(attributeMatches))
```

Reply:
Fixed in https://github.com/SumoLogic/sumologic-otel-collector/pull/257/commits/aba0f0da5d5ee2662dc1044873b883e9ae914f20